libs/x-content/src/test/java/org/elasticsearch/xcontent/support/filtering/FilterPathTests.java

```diff
@@ -338,4 +338,33 @@ public void testNoMatchesFilter() {
         assertFalse(filterPath.matches(randomAlphaOfLength(10), nextFilters));
         assertEquals(nextFilters.size(), 0);
     }
+
+    public void testDotInFieldName() {
+        // FilterPath match
+        FilterPath[] filterPaths = FilterPath.compile(singleton("foo"));
+        List<FilterPath> nextFilters = new ArrayList<>();
+        assertTrue(filterPaths[0].matches("foo.bar", nextFilters, true));
+        assertEquals(nextFilters.size(), 0);
+
+        // FilterPath not match
+        filterPaths = FilterPath.compile(singleton("bar"));
+        assertFalse(filterPaths[0].matches("foo.bar", nextFilters, true));
+        assertEquals(nextFilters.size(), 0);
+
+        // FilterPath equals to fieldName
+        filterPaths = FilterPath.compile(singleton("foo.bar"));
+        assertTrue(filterPaths[0].matches("foo.bar", nextFilters, true));
+        assertEquals(nextFilters.size(), 0);
+
+        // FilterPath longer than fieldName
+        filterPaths = FilterPath.compile(singleton("foo.bar.test"));
+        assertFalse(filterPaths[0].matches("foo.bar", nextFilters, true));
+        assertEquals(nextFilters.size(), 1);
+        nextFilters.clear();
+
+        // partial match
+        filterPaths = FilterPath.compile(singleton("foo.bar.test"));
+        assertFalse(filterPaths[0].matches("foo.bar.text", nextFilters, true));
+        assertEquals(nextFilters.size(), 0);
+    }
```

Comment:
I think it's worth taking some of the tests from [my PR](https://github.com/elastic/elasticsearch/pull/83148/files#diff-f1ab9e67867791ff2768917c0f445128aaac5886ee3c8eb15322db370b5f39e7) into this one. Just to test the filtering "above" this layer.

Reply:
Thanks , that's what I intended to do.

Reply:
Nice!